internal/probe.go

```diff
@@ -76,19 +79,40 @@ func (p Probe) Do(ctx context.Context) error {
 					p.Logger.Debug(
 						"New protocol", "count", count, "protocol", proto,
 					)
-					rhost, extra, err := proto.Probe("")
-					report := Report{
-						ProtocolID: proto.String(),
-						Time:       time.Since(start),
-						Error:      err,
-						RHost:      rhost,
-						Extra:      extra,
+
+					if len(p.Addrs) == 0 {
+						// Original behavior
+						rhost, extra, err := proto.Probe("")
+						report := Report{
+							ProtocolID: proto.String(),
+							Time:       time.Since(start),
+							Error:      err,
+							RHost:      rhost,
+							Extra:      extra,
+						}
+						p.Logger.Debug(
+							"Sending report back",
+							"count", count, "report", report,
+						)
+						p.ReportCh <- &report
```

Comment:
I would add a return here to remove the `else`. Only to avoid the extra identation.